### DISCUSSION TITLE
20240516 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#94] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x
- Iván Santana(@ivan9989-pkj) - 
- Ismael Torrado(@Lepisma07) - x
- Borja (@BorjaPractica) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

Configuration Overhaul:

- [x] https://github.com/torrust/torrust-tracker/issues/851
- [x] https://github.com/torrust/torrust-tracker/issues/852
- [x] https://github.com/torrust/torrust-tracker/issues/855
- [x] https://github.com/torrust/torrust-tracker/issues/856
- https://github.com/torrust/torrust-tracker/pull/808 
- New proposal for toml config file (version 2): https://github.com/torrust/torrust-tracker/discussions/853

Key points in configuration overhaul:

- We use [Figment](https://crates.io/crates/figment).
- You don't need to provide all config options. You only provide the ones you want to override from default values.
- All config values can be overridden via environment variables.
- Env varas were renamed:
  - To follow Figment conventions
  - To give them better names
  - To allow using other config file formats like JSON, in the future.

Review timeouts (all issues related to timeouts on the server side are fixed):

- [x] https://github.com/torrust/torrust-tracker/issues/612
- [x] https://github.com/torrust/torrust-tracker/issues/613

**NOTE**: The server timeout for the first request after opening the connection can be improved in the future by returning a "408 Request Timeout" message like ActixWeb. The current solution only closes the connection without sending a message to the client, but it seems to be an acceptable and regular practice.

Update dependencies.

### Torrust Index

New issues:

- [ ] https://github.com/torrust/torrust-index/issues/580

Issues:

- [ ] https://github.com/torrust/torrust-index/issues/426 (**WIP** but almost finished)

PR:

- https://github.com/torrust/torrust-index/pull/579

Update dependencies.

### Torrust Index GUI

Issues:

- [x] https://github.com/torrust/torrust-index-gui/issues/553

Update dependencies.

### Torrust Website

PR Review:

- https://github.com/torrust/torrust-website/pull/177

### Torrust Demo

Issues:

- [x] https://github.com/torrust/torrust-demo/issues/5

**Comment by mario-nt** [comment]
*****Meeting*****

### Index
	
- https://github.com/torrust/torrust-index/issues/448
- https://github.com/mario-nt/torrust-index/tree/casbin-test

### Training/Research

- Tidy first? book
- RwLocks and Tokio async Rwlocks 101
- Debugging with Rust analyzer

### Other

**Reviewed:**

- https://github.com/torrust/torrust-index-gui/issues/448#event-12574721322
- https://joinpeertube.org/
- https://github.com/torrust/torrust-index-gui/issues/428
